src/main/java/world/bentobox/parkour/listeners/CourseRunnerListener.java

```diff
@@ -113,27 +114,32 @@ public void onVisitorFall(EntityDamageEvent e) {
         player.playEffect(EntityEffect.ENTITY_POOF);
         player.setVelocity(new Vector(0, 0, 0));
         player.setFallDistance(0);
-        player.teleport(parkourRunManager.checkpoints().get(player.getUniqueId()));
+        Location checkpointLocation = parkourRunManager.checkpoints().get(player.getUniqueId());
+        checkpointLocation = checkpointLocation.clone().add(0.5, 0, 0.5);
+        parkourRunManager.currentlyTeleporting().add(player.getUniqueId());
+        player.teleport(checkpointLocation);
+        parkourRunManager.currentlyTeleporting().remove(player.getUniqueId());
```

Comment:
The removal would then happen after the completable future has completed, e.g., `teleportAsync( entity, location, cause).thenAccept(b -> parkourRunManager.currentlyTeleporting().remove(player.getUniqueId()));`
or similar.